src/plugins/intel_cpu/src/nodes/kernels/x64/brgemm_kernel.cpp

```diff
@@ -36,28 +36,41 @@ BrgemmKernel::BrgemmKernel(size_t M,
     // blocking M
     M_blk = matmulOptimalM;
     M_tail = M % M_blk;
-    brgVnniFactor = 4 / inType.size();
 
-    if (inType != ov::element::bf16 && inType != ov::element::f32)
-        THROW_ERROR("brgemm kernel only supports bf16, f32");
-    bool is_bf16 = inType == ov::element::bf16;
-    if (is_bf16 && !mayiuse(avx512_core_bf16))
+    if (inType != ov::element::bf16 && inType != ov::element::f32 && inType != ov::element::f16)
+        THROW_ERROR("brgemm kernel only supports fp16, bf16, f32");
+    bool is_f32 = inType == ov::element::f32;
+    if (inType == ov::element::bf16 && !mayiuse(avx512_core_bf16))
         THROW_ERROR("brgemm bf16 kernel could only be used above avx512_bf16");
+    // TODO: amx_f16
+    if (inType == ov::element::f16 && (!mayiuse(avx512_core_fp16)))
+        THROW_ERROR("brgemm f16 kernel could only be used with avx512_core_fp16/avx2_vnni_2");
```

Comment:
please remove the avx2_vnni_2 since it's not be support yet

Reply:
Done.